scalyr_agent/log_processing.py

```diff
@@ -569,9 +649,13 @@ def readline(self, current_time=None):
             return LogLine(line=b"")
 
         # Keep our underlying buffer of bytes filled up.
-        if self.__buffer is None or (
-            available_buffer_bytes < self.__max_line_length
-            and more_file_bytes_available
+        if (
+            self.__buffer is None
+            or more_file_bytes_available
+            and (
+                available_buffer_bytes < self.__max_line_length
+                or not self.__have_buffer_for_fragment_position(available_buffer_bytes)
+            )
```

Comment:
Thanks for that.  Ok, I've redid it a little bit and I think it is more readable now.  I also realized that the `self.__buffer is None` check is superfluous because the second condition (whether or not we have at least `max_line_size`  bytes available in the buffer) will cover that case as well.